lib/library/AssignLibraryPlugin.js

```diff
@@ -296,25 +301,41 @@ class AssignLibraryPlugin extends AbstractLibraryPlugin {
 				)
 			: "";
 		const result = new ConcatSource(source);
+		const hooks = getModuleFederationPlugin().getCompilationHooks(compilation);
```

Comment:
i need to add this to all the other library plugins as well, so that i can get the rendering of the exports field correctly and can implement it outside the wrapper again

Reply:
Maybe we need one hook where you wrap it in your code in your plugin, because now we have a tight connection between the library and the federation of modules, I don't think this is quite the right design decision

Reply:
Any ideas or should I look into this further and help you?

Reply:
If you can help me with relocating this to the right place. That would be very helpful. I'm happy to do the heavy lifting - but I wanted to send this WIP PR for exactly this reason. Help me get things in the right spot. The PR shows in general what I must change to make it work. I don't believe this is the right way, there should be a less complex way to hook into all the places correctly. Some assistance to get me in the right direction would be immensely helpful

Reply:
One idea i had was to split the export code to left and right. So there's something like render export. On left is the var or runtime loops to assign to the var or object and on the right is the equals part that I can wrap. 


Like LEFT = RIGHT 

Exanple module.exports = webpack_require stuff